strongbox-security/strongbox-authentication-registry/src/test/java/org/carlspring/strongbox/authentication/ExternalUserDetailsServiceTest.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@ActiveProfiles({ "test", "ExternalUserDetailsServiceTestConfig" })
+@TestPropertySource(properties = { "strongbox.config.file.authentication.providers=classpath:eudst-strongbox-authentication-providers.xml",
+                                   "strongbox.authentication.providers.yaml=classpath:/etc/conf/eudst-strongbox-authentication-providers.yaml",
+                                   "users.external.cache.seconds=3600",
+                                   "strongbox.users.config.yaml=classpath:/etc/conf/eudst-strongbox-security-users.yaml" })
+@ContextConfiguration(classes = TestConfig.class)
+public class ExternalUserDetailsServiceTest
+{
+
+    private static final String TEST_USER = "test-user";
+
+    @Value("${users.external.cache.seconds}")
+    private int externalUsersInvalidateSeconds;
+
+    @Inject
+    @OrientDb
+    private UserService orientDbUserService;
+
+    @Inject
+    @Yaml
+    private UserService yamlUserService;
+
+    @Inject
+    private StrongboxUserManager strongboxUserManager;
+
+    @Inject
+    private UserDetailsService userDetailsService;
+
+    @Inject
+    private PasswordEncoder passwordEncoder;
+
+    @PersistenceContext
+    protected EntityManager entityManager;
+
+    @Test
+    @Transactional
+    public void testExternalUserCache()
+    {
+        // Check that there is no external user cached
+        assertNull(strongboxUserManager.findByUsername(TEST_USER));
+
+        // Load and cache external user
+        assertNotNull(userDetailsService.loadUserByUsername(TEST_USER));
+
+        // Check that external user cached
+        User externalUser = strongboxUserManager.findByUsername(TEST_USER);
+        assertNotNull(externalUser);
+
+        // Check that external user can't be modyfied
+        User externalUserToSave = externalUser;
+        assertThrows(IllegalStateException.class, () -> orientDbUserService.save(externalUserToSave),
+                     () -> "Can't modify external users.");
+
+        // Update external password
+        String newPassword = UUID.randomUUID().toString();
+        UserDto userToUpdate = new UserDto();
+        userToUpdate.setUsername(TEST_USER);
+        userToUpdate.setPassword(newPassword);
+        yamlUserService.updateAccountDetailsByUsername(new EncodedPasswordUser(userToUpdate, passwordEncoder));
+
+        // Invalidate user cache
```

Comment:
` // Invalidate user cache` how below part of code invalidates the user cache ? Is this comment valid ?

Reply:
You meant the `database storage` saying `cache` ?

Reply:
yes, this commend is valid
and it mean exactly the "external user cache" which is stored in DB :)

I've added some more javadocs within `StrongboxExternalUsersCacheManager`, please have a look, maybe it can make things a bit more clear

Also please share your thoughts if we need more clarifications